test/controllers/redirect.controller.test.ts

```diff
@@ -264,10 +281,13 @@ describe('RedirectController', () => {
         .get(PAYMENT_CALLBACK_URI)
         .query({
           state: 'ABC123',
-          status: PaymentStatus.CANCELLED
+          status: PaymentStatus.CANCELLED,
+          ref: '123456'
         })
         .expect(MOVED_TEMPORARILY)
         .expect('Location', ERROR_URI)
+
+      verify(session.setDissolutionSession(anything(), anything())).once()
```

Comment:
i think one test per public method should ensure that the application reference number was actually saved in session. This can be achieved using ArgCaptor